src/embind/embind_gen.js

```diff
@@ -182,24 +187,24 @@ var LibraryEmbind = {
       out.push('};\n');
     }
 
-    printJs(out) {
+    printJs(emitted, out) {
       out.push(`// class ${this.name}\n`);
       if (this.constructors.length) {
         out.push(`// constructors\n`);
         for (const construct of this.constructors) {
-          construct.printJs(out);
+          construct.printJs(emitted, out);
```

Comment:
Rather than passing this around maybe we can just use some global state such as `LibraryEmbind.emitted`?